Resources/doc/Sync/StorageManager/storage_manager.rst

```diff
@@ -0,0 +1,28 @@
+==============
+StorageManager
+==============
+
+All commands required for the storage are executing by ``StorageManager``. It has 4 functions that need to be implemented by any type of
+``StorageManager``:
+
+- to create a storage;
+- to add a record to the storage or update existing one with new date and time;
+- remove a record from storage;
+- to get data from the storage.
+
+Every function can work with multiple shops. ``MysqlStorageManager`` is a default manager.
+
+Adding new records to the storage
+-------------------------------------
+
+After every INSERT operation, ``StorageManager`` checks if DELETE operation exists. It is required, because any
```

Comment:
Consider rephrasing:

``StorageManager`` checks if DELETE operation exists after every INSERT operation, because any operation before DELETE is meaningless.